pkgs/development/python-modules/yappi/default.nix

```
@@ -1,18 +1,22 @@
-{ lib, buildPythonPackage, fetchPypi, isPy27, nose }:
+{ lib, buildPythonPackage, fetchFromGitHub, gevent, isPy27, python }:
 
 buildPythonPackage rec {
   pname = "yappi";
-  version = "1.2.5";
+  version = "1.3.0";
   disabled = isPy27; # invalid syntax
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "ad5fa4caf2859e480ffc4ec3e85615a6f7dea852c8f035f2db723f824ed4ba11";
+  src = fetchFromGitHub {
+    owner = "sumerc";
+    repo = pname;
+    rev = "30f94024a0e2e4fa21c220de6a0dc97b4cb2c319";
+    sha256 = "1kvwl3y3c2hivf9y2x1q1s8a2y724iwqd1krq6ryvsbg3inyh8qw";
   };
 
   patches = [ ./tests.patch ];
-
-  checkInputs = [ nose ];
+  checkInputs = [ gevent ];
+  checkPhase = ''
+    ${python}/bin/python run_tests.py
```

to support alternate python implementations
```suggestion
    ${python.interpreter} run_tests.py
```